proposals/p2360.md

```diff
@@ -0,0 +1,598 @@
+# Types are values of type `type`
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/2360)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Abstract](#abstract)
+-   [Problem](#problem)
+    -   [Tuples behave inconsistently](#tuples-behave-inconsistently)
+    -   [Qualified access to constraint members is inconsistent](#qualified-access-to-constraint-members-is-inconsistent)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Details](#details)
+    -   [Terminology](#terminology)
+    -   [`type` and `Core.Type`](#type-and-coretype)
+    -   [Compound member access](#compound-member-access)
+        -   [Instance binding](#instance-binding)
+        -   [Impl lookup](#impl-lookup)
+        -   [Examples](#examples)
+    -   [Tuple and struct literals](#tuple-and-struct-literals)
+    -   [Name lookup into values with constrained types](#name-lookup-into-values-with-constrained-types)
+-   [Rationale](#rationale)
+    -   [Teachability](#teachability)
+-   [Alternatives considered](#alternatives-considered)
+    -   [Alternative member access rules](#alternative-member-access-rules)
+    -   [Alternative terminology](#alternative-terminology)
+    -   [`Core.Type` is an empty named constraint](#coretype-is-an-empty-named-constraint)
+    -   [Write tuple types differently than tuples](#write-tuple-types-differently-than-tuples)
+
+<!-- tocstop -->
+
+## Abstract
+
+Define a "type" to be a value of type `type`. Contexts expecting a type perform
+an implicit conversion to `type`. Values like `()`, `(i32, i32)`, and `{}` are
+no longer types, but instead implicitly convert to `type`. Values of interface
+or constraint type (now called "facets") are similarly not formally types but
+implicitly convert to `type`.
+
+The practical impact to the language semantics is relatively small. This
+proposal primarily establishes clearer, more principled terminology and design
+concepts with a few narrow improvements to the design's consistency.
+
+## Problem
```

Comment:
Another problem that was raised [on 2022-12-07](https://docs.google.com/document/d/1gnJBTfY81fZYvI_QXjwKk1uQHYBNHGqRLI2BS_cYYNQ/edit?resourcekey=0-ql1Q1WvTcDvhycf8LbA9DQ#heading=h.qu6e01wbzrhz):
> [We] don't want `i32 as Sortable` and `i32 as Hashable` to be different _as types_, even though we want them to be different